lib/spack/spack/test/build_systems.py

```diff
@@ -181,3 +181,41 @@ def test_none_is_allowed(self):
         assert '--without-bar' in options
         assert '--without-baz' in options
         assert '--no-fee' in options
+
+
+@pytest.mark.usefixtures('config', 'mock_packages')
+class TestCMakePackage(object):
+
+    def test_define(self):
+        s = Spec('cmake-client')
+        s.concretize()
+        pkg = spack.repo.get(s)
+
+        for cls in (list, tuple):
+            arg = pkg.define('MULTI', cls(['right', 'up']))
+            assert arg == '-DMULTI:STRING=right;up'
+
+        arg = pkg.define('ENABLE_TRUTH', False)
+        assert arg == '-DENABLE_TRUTH:BOOL=OFF'
+        arg = pkg.define('ENABLE_TRUTH', True)
+        assert arg == '-DENABLE_TRUTH:BOOL=ON'
+
+        arg = pkg.define('SINGLE', 'red')
+        assert arg == '-DSINGLE:STRING=red'
+
+    def test_define_from_variant(self):
+        s = Spec('cmake-client multi=up,right ~truthy single=red')
+        s.concretize()
+        pkg = spack.repo.get(s)
+
+        arg = pkg.define_from_variant('MULTI')
+        assert arg == '-DMULTI:STRING=right;up'
```

Comment:
Should `CMakePackage.define` sort the input values, or should it respect the order defined by the user? I don't see sorting happening in `define` right now (although it so happens that the result is alphabetically ordered) so I think it may be useful to enforce that there - otherwise this test may be fragile. If the order shouldn't matter, then another option is to deconstruct the string in this test.

Reply:
Aha, good catch. I will do that. (I guess I assumed multi-variant values were already sorted since they always appear that way in a spec?)

Reply:
That's definitely true that multivalued variants are sorted within the spec (e.g. to ensure that the DAG hash is repeatable). So come to think of it there is no need!

Reply:
Well, I added the code in there anyway. Shouldn't hurt, right?